src/styles/_typography.scss

```diff
@@ -15,10 +19,21 @@ a[href] {
     text-decoration: none;
     outline: var(--gcds-border-width-md) solid var(--gcds-focus-background);
   }
-}
 
-a[href]:not([class*="link-inherit"]):not([class*="link-light"]):not(:focus):not(:hover) {
-  color: var(--gcds-link-default);
+  // Exceptions
+  .text-light & {
+    color: currentColor;
+    &:hover {
+      color: currentColor;
+    }
+  }
+
+  .bg-blue-900 & {
+    color: var(--gcds-link-light);
+    &:hover {
+      color: var(--gcds-link-light);
+    }
+  }
```

Comment:
This code only covers one text colour and one bg colour and isn't very future-proof. I think it's better to target the exception classes as I did above and then use the `link-inherit` class in the html to set it to the current colour. Otherwise, we'll need to keep track of all these different text and bg colours.